res/controllers/Roland_DJ-505-scripts.js

```diff
@@ -1638,26 +1634,23 @@ DJ505.SavedLoopMode = function(deck, offset) {
             }.bind(this);
         },
         stopBlinking: function() {
-            if (this.blinkTimer !== 0) {
-                engine.stopTimer(this.blinkTimer);
-                this.blinkTimer = 0;
+            if (this.connections[2] !== undefined) {
+                this.connections[2].disconnect();
+                delete this.connections[2];
             }
         },
         output: function(value, _group, _control) {
             this.stopBlinking();
             if (value === 2) {
-                this.blinkTimer = engine.beginTimer(
-                    this.blinkTimeout, function() {
-                        var colorValue = this.colorMapper.getValueForNearestColor(
-                            engine.getValue(this.group, this.colorKey));
-                        if (this.blinkStateOn) {
-                            this.send(colorValue + DJ505.PadColor.DIM_MODIFIER);
-                            this.blinkStateOn = false;
-                        } else {
-                            this.send(colorValue);
-                            this.blinkStateOn = true;
-                        }
-                    }.bind(this));
+                this.connections[2] = engine.makeConnection("[Master]", "indicator_250millis", function(value, _group, _control) {
+                    var colorValue = this.colorMapper.getValueForNearestColor(
+                        engine.getValue(this.group, this.colorKey));
+                    if (value) {
+                        this.send(colorValue);
+                    } else {
+                        this.send(colorValue + DJ505.PadColor.DIM_MODIFIER);
+                    }
+                }.bind(this));
```

Comment:
Now that we have ES7 support, we should phase out the function expression + `.bind(this) ` pattern and use arrow functions instead IMO.

Reply:
Arrow functions are harder to read IMHO, but that aside, when I apply this patch:

```diff
+++ b/res/controllers/Roland_DJ-505-scripts.js
@@ -1642,7 +1642,7 @@ DJ505.SavedLoopMode = function(deck, offset) {
         output: function(value, _group, _control) {
             this.stopBlinking();
             if (value === 2) {
-                this.connections[2] = engine.makeConnection("[Master]", "indicator_250millis", function(value, _group, _control) {
+                this.connections[2] = engine.makeConnection("[Master]", "indicator_250millis", (value, _group, _control) => {
                     var colorValue = this.colorMapper.getValueForNearestColor(
                         engine.getValue(this.group, this.colorKey));
                     if (value) {
@@ -1650,7 +1650,7 @@ DJ505.SavedLoopMode = function(deck, offset) {
                     } else {
                         this.send(colorValue + DJ505.PadColor.DIM_MODIFIER);
                     }
-                }.bind(this));
+                });
             } else if (value === 1) {
                 var colorValue = this.colorMapper.getValueForNearestColor(
                     engine.getValue(this.group, this.colorKey));
```
I get:
```
Uncaught exception at line 1649 in file file:///home/jan/Projects/mixxx/res/controllers/Roland_DJ-505-scripts.js.

Exception:
  TypeError: Property 'send' of object [object Object] is not a function

Backtrace:
@file:///home/jan/Projects/mixxx/res/controllers/Roland_DJ-505-scripts.js:1649
```

Reply:
Thats strange. I'll try to reproduce.

Reply:
> Arrow functions are harder to read IMHO

I kind off disagree, they're just like closures from most other functions (C++ lambdas, Rust `Fn`, etc). And they get rid of the `.bind(this)` that most people without indepth JS knowledge don't understand.

Reply:
> > Arrow functions are harder to read IMHO
> 
> I disagree, they're just like closures from most other functions (C++ lambdas, Rust `Fn`, etc). And they get rid of the `.bind(this)` that most people without indepth JS knowledge don't understand.

I'd argue that people without indepth JS knowledge wouldn't even recognize that `foo => bar` is a function. I think `function(foo) { return bar; }` is much clearer to a beginner. Just my 2 cents.

Reply:
it depends, I think function expressions are being thought quite early now especially with the rising popularity of functional programming and APIs that use callbacks.

Reply:
I tried to reproduce the issue but I was unable to:
```javascript
let Foo = function () {
    components.Button.call(this);
}
Foo.prototype = new components.Button({
    init: function() {
        engine.makeConnection("[Channel1]", "play", (value, group, control) => {
            print(typeof this.send);
        });
    }
});

var MyController = new Foo();

```
Obviously the example is a bit atypical but the point is the same.

Reply:
I don't really care if its an arrow function or a plain old function expression in this. This is just your mapping overall. Though I'd like to establish the use of arrow functions for any new code in the future with the ES6 rewrite of components JS.

Reply:
Ok, let's just merge this then, and I'll try to make a PR to use arrow functions later.